### PR TITLE
Welcome Screen Complete

### PR DESCRIPTION
**Pull request by sivaShilpa**
### Done

1. Edited README to add the windows command to open the JSON settings file in the Visual Studio code and also the git flow.
2. Created components and screens Folder
3. Moved the components and screens folder under the src folder
4. Added Button.tsx and Button.spec.tsx to the Components Folder
5. Added WelcomeScreen.tsx and WelcomeScreen.spec.tsx to screens folder
6. Added font files to assets folder. 
7. Complete the Landing Page.
8. Wrote the test files.
9. Resolved all changes requested except for the test files 

### Notes

- Font Georgia for the title BeeSmart is not open source so used another similar font NotoSerif
- Please find the demo here. https://jam.dev/c/5a14d628-f634-4087-a9bf-b445adb67680

### CODE REVIEW

File: src/Contributors.tsx [hunk 1]
```diff
@@ -2,26 +2,27 @@ import React from 'react';
 import { StyleSheet, Text, ScrollView } from 'react-native';
 
 const Contributors = () => {
-    const contributorsList = ['Melisa Im', 'Monica Lam', 'Linden Chiu'];
+  const contributorsList = ['Melisa Im', 'Monica Lam', 'Linden Chiu', 'Shilpa Sivarajan'];
```
Comment (lindentree) [inline src/Contributors.tsx:5]: Please move this change to its own PR; each PR should be for one feature/change.

File: src/screens/WelcomeScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+const landingPage = () => {
```
Comment (lindentree) [inline src/screens/WelcomeScreen.tsx:3]: In the future, please do not PR until code is functional and ready for acceptance. This screen needs to be accessible and render in the app, even with placeholder text/images. Also please add functional code to the empty files or remove them from the PR.

**Review by lindentree — CHANGES_REQUESTED** [review]

File: src/screens/WelcomeScreen.tsx (hunk 2 above)
Comment (MelisaIm) [reply src/screens/WelcomeScreen.tsx:3]: It's okay to open a PR with unfinished work if it's a `draft` PR :)

Comment (MelisaIm) [reply src/screens/WelcomeScreen.tsx:3]: I agree with, Linden, that if the purpose of this change is to add a screen, that screen should be accessible for validation :) 

Good job so far!

File: src/Contributors.tsx (hunk 1 above)
Comment (sivaShilpa) [reply src/Contributors.tsx:5]: Moved this to a new PR from a new branch shilpa/addContributor

File: README.md [hunk 3]
```diff
@@ -9,33 +9,40 @@ To run the app locally, clone this repo, cd into the repo folder and then run th
...
 
     After this, go to the Contributors.tsx file, and add your name to the contributorsList array variable. Now when you run `npm test` again, all tests should pass. This is a simpler form of [Red-Green-Refactor.](https://www.codecademy.com/article/tdd-red-green-refactor) All features and code you add in the future will be expected to have adequate test coverage, or they will be turned back!
 
-    You can then verify that your name was added properly by opening the running app and going to the Contributors tab via the bottom navigation bar. After that, open a pull request to main so an owner can approve it. Once it is approved, merge it, and you are officially part of the Swarm beekeeping project!
\ No newline at end of file
+    You can then verify that your name was added properly by opening the running app and going to the Contributors tab via the bottom navigation bar. After that, open a pull request to main so an owner can approve it. Once it is approved, merge it, and you are officially part of the Swarm beekeeping project!
+
+### Git Flow
```
Comment (lindentree) [inline README.md:44]: -> Git Flow When Another PR Has Been Merged Ahead Of Yours

File: src/screens/WelcomeScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+  function handleSignUp(): void {
+    navigation.navigate('SignUpScreen');
+  }
+
+  return (
+    <View style={styles.container}>
+      <Image
+        source={{
+          uri: 'https://s3-alpha-sig.figma.com/img/ceba/f0c9/1d52df65bfda2146d8a56199f05e993c?Expires=1716768000&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4&Signature=PbvjzmBR0H1OEIP7kYo8OhAQSI9RAQh9XaiXD5-6PQFNEONVvgAT7j2FZoDX5LV-Ni0jhpLQe2Fb1Igm0Trz5uPsPNcFVpj3C4JvWF3sGYOVuWvHD50bsJt8EwMAugCSWSiWLiW4WXyRuSO7jvgePgeBHytcafC1byPQqRLxIUiAUpOjyqPq1yJhk1Ax6qzKrs9G5zLY11bY2MsevVSvI5tIWrHhMSsz2W2kP61elVfkbmuMHAQTaHMHUvCtWjxifMdcJiTIblt~356FjxQKgSSsuuWsNHOiA49Q2yFf~HgLQ-f1XQ8F4kX5p7lI3UiiFE6nk6jP15iZqfHk8pMWPw__',
```
Comment (lindentree) [inline src/screens/WelcomeScreen.tsx:30]: why point to a remote URI? unless we're memory constrained, it would probably be better to have the image file in the assets folder.

File: src/screens/__snapshots__/LoginScreen.spec.tsx.snap [hunk 5]
```diff
@@ -0,0 +1,7 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Contributors should render the Welcome Screen 1`] = `
+<div>
```
Comment (lindentree) [inline src/screens/__snapshots__/LoginScreen.spec.tsx.snap:4]: These snapshots don't look correct. Can they be regenerated?

File: src/Contributors.tsx [hunk 6]
```diff
@@ -2,26 +2,27 @@ import React from 'react';
 import { StyleSheet, Text, ScrollView } from 'react-native';
 
 const Contributors = () => {
-    const contributorsList = ['Melisa Im', 'Monica Lam', 'Linden Chiu'];
+  const contributorsList = ['Melisa Im', 'Monica Lam', 'Linden Chiu'];
```
Comment (lindentree) [inline src/Contributors.tsx:5]: this looks like your new contributor PR is not merged in properly

File: react-native.config.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export default {
+  project: {
+    ios: {},
+    android: {},
+  },
+  assets: ['./assets/fonts'],
```
Comment (lindentree) [inline react-native.config.ts:6]: assets/fonts should not be a top level folder. make fonts a subdirectory and place the font folders inside

File: src/App.tsx [hunk 8]
```diff
@@ -28,27 +34,23 @@ const App = () => {
...
-
             // You can return any component that you like here!
             return <Ionicons name={iconName} size={size} color={color} />;
           },
           tabBarActiveTintColor: 'tomato',
           tabBarInactiveTintColor: 'gray',
         })}
       >
-        <Tab.Screen name="Home" component={HomeScreen} />
+        <Tab.Screen name="Home" component={HomeStack} options={{ headerShown: false }} />
```
Comment (lindentree) [inline src/App.tsx:48]: Nicely done!

**Review by lindentree — CHANGES_REQUESTED** [review]

File: src/screens/WelcomeScreen.tsx (hunk 4 above)
Comment (sivaShilpa) [reply src/screens/WelcomeScreen.tsx:30]: I am waiting for Monica to share the google drive of pics. Once I get that I will change them.

File: src/screens/__snapshots__/LoginScreen.spec.tsx.snap (hunk 5 above)
Comment (sivaShilpa) [reply src/screens/__snapshots__/LoginScreen.spec.tsx.snap:4]: I will have to rewrite the test files. I will get them done by Friday.

File: src/Contributors.tsx (hunk 6 above)
Comment (sivaShilpa) [reply src/Contributors.tsx:5]: I didn't merge the branch shilpa/addContributor to this branch. It was merged to main and then I deleted that branch. The change is in the main branch though.

File: react-native.config.ts (hunk 7 above)
Comment (sivaShilpa) [reply react-native.config.ts:6]: I am not sure how to do this. Is it like below?
```
fonts: ['./assets/fonts] , 
```
Else please help me with this on Saturday meeting.

File: src/App.tsx (hunk 8 above)
Comment (sivaShilpa) [reply src/App.tsx:48]: Thank you!

File: src/Contributors.tsx (hunk 6 above)
Comment (lindentree) [reply src/Contributors.tsx:5]: You have to merge main into this PR branch as described in the README

File: src/screens/WelcomeScreen.tsx (hunk 4 above)
Comment (lindentree) [reply src/screens/WelcomeScreen.tsx:30]: Just as a note for future, exposing project URLs like this in code is bad practice, ask for placeholder assets or use an environment variable

File: src/screens/__snapshots__/LoginScreen.spec.tsx.snap (hunk 5 above)
Comment (lindentree) [reply src/screens/__snapshots__/LoginScreen.spec.tsx.snap:4]: No rush

File: react-native.config.ts (hunk 7 above)
Comment (lindentree) [reply react-native.config.ts:6]: The path reference is correct, but I think you named the folder ‘assets/fonts’ instead of nesting it properly. You don’t need to make any changes here, you just need to create the fonts subdirectory inside assets and rename the top level. You may need to adjust the font references elsewhere afterwards.

File: src/screens/__snapshots__/LoginScreen.spec.tsx.snap (hunk 5 above)
Comment (sivaShilpa) [reply src/screens/__snapshots__/LoginScreen.spec.tsx.snap:4]: I will need more time to write the correct jest test files. The ones I am rewriting are failing the test and I am unsure how to correct them. I will go through the jest documents and get them done by the end of next week.

Comment (lindentree) [reply src/screens/__snapshots__/LoginScreen.spec.tsx.snap:4]: As long as your tests run correctly locally, I would suggest removing and not committing these snapshot files, so we can approve the rest of this PR. Pending @MelisaIm approval

Comment (sivaShilpa) [reply src/screens/__snapshots__/LoginScreen.spec.tsx.snap:4]: Resolved the test file issues. The snapshots are looking good to me now. Please take a look.

File: src/Contributors.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lindentree) [reply src/Contributors.tsx:5]: Still need to merge in main, this method to resolve creates too many potential issues. Everything else LGTM

**Review by lindentree — CHANGES_REQUESTED** [review]

**Comment by lindentree** [comment]
Closing as redundant to SignUp Screen Complete #14

File: src/Contributors.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sivaShilpa) [reply src/Contributors.tsx:5]: I merged main and it says it is already up to date

[the remark this answers is not in the document]
Comment (lindentree) [reply src/Contributors.tsx:5]: Can you run ‘git rebase main’ and provide the output?

[the remark this answers is not in the document]
Comment (lindentree) [reply src/Contributors.tsx:5]: Also just as a sanity check, please make sure you ran git pull — rebase in main and have the latest.
